test/extensions/stats_sinks/common/statsd/udp_statsd_test.cc

```diff
@@ -39,15 +39,10 @@ INSTANTIATE_TEST_SUITE_P(IpVersions, UdpStatsdSinkTest,
 
 TEST_P(UdpStatsdSinkTest, InitWithIpAddress) {
   NiceMock<ThreadLocal::MockInstance> tls_;
-  NiceMock<Stats::MockMetricSnapshot> snapshot; // UDP statsd server address.
-  Network::Address::InstanceConstSharedPtr server_address =
-      Network::Utility::parseInternetAddressAndPort(
-          fmt::format("{}:8125", Network::Test::getLoopbackAddressUrlString(GetParam())));
-  UdpStatsdSink sink(tls_, server_address, false);
-  int fd = sink.getFdForTests();
-  EXPECT_NE(fd, -1);
-
-  // Check that fd has not changed.
+  NiceMock<Stats::MockMetricSnapshot> snapshot;
+  Network::Test::UdpSyncPeer server(GetParam());
```

Comment:
Does this test fail when writer use connect() and send()? I don't fully understand the issue https://github.com/envoyproxy/envoy/issues/8911, but if this PR fixes it, it's better to add a regression test.

Reply:
The original issue is that the UDP server is not listening when connect happens. By not using connect and using sendmsg with an address specified each time, we avoid this issue. I suppose I could regression test this specific case, but IMO it's not worth the effort. If you think it's really worth it I can do it though.

Reply:
> The original issue is that the UDP server is not listening when connect happens. By not using connect and using sendmsg with an address specified each time, we avoid this issue. 

I think UDP socket connect() does nothing other than store the destination address for later send()/write(). Does it matter if the UDP server is listening or not? 

> I suppose I could regression test this specific case, but IMO it's not worth the effort. If you think it's really worth it I can do it though.

I would prefer a regression test if it not too complicated. Since the issue happens with UDS, it seems that we don't have test to cover such type of socket.

Reply:
Good eye, I didn't look at the individual report carefully enough. Yes let me add a test specifically for UDS for this.